### PR TITLE
Feat: refactor project structure

### LINKED ISSUE #130: Refactor project structure.
1. Move most of the code under "internal".
2. Move terratag executable under "cmd".

### PR DESCRIPTION
**Pull request by TomerHeber**
closes #130 

terratag executable moved to a directory called "cmd/terratag".
In the main directory there is a package called "terratag" the only files in there is the terratag.go library and terratag_test.go library tester.
the rest of the go files (and their directories) have been moved under "internal" directory.

Here is some information about internal directory:
https://go.dev/doc/go1.4#internalpackages

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
       - name: Build project
         run: |
           mkdir out
-          CGO_ENABLED=0 go build -a -installsuffix cgo -o ./out/terratag/terratag
+          CGO_ENABLED=0 go build ./cmd/terratag -a -installsuffix cgo -o ./out/terratag/terratag
```
Comment (TomerHeber) [inline .github/workflows/release.yml:50]: This change I cannot test.
When releasing a new terratag version need to make sure I didn't break anything in the release process.

File: terratag.go [hunk 2]
```diff
@@ -7,19 +7,34 @@ import (
...
+	"github.com/env0/terratag/internal/tag_keys"
+	"github.com/env0/terratag/internal/tagging"
+	"github.com/env0/terratag/internal/terraform"
+	"github.com/env0/terratag/internal/tfschema"
+	"github.com/env0/terratag/internal/utils"
 	"github.com/hashicorp/hcl/v2/hclwrite"
 )
 
-func Terratag(args Args) {
+type counters struct {
```
Comment (TomerHeber) [inline terratag.go:23]: This is the only place counters is required.
Moved it to here.

File: terratag_test.go [hunk 3]
```diff
@@ -13,13 +13,13 @@ import (
...
 	"github.com/bmatcuk/doublestar"
-	"github.com/env0/terratag/cli"
+	"github.com/env0/terratag/internal/cli"
 	. "github.com/onsi/gomega"
 	"github.com/otiai10/copy"
 	"github.com/spf13/viper"
 )
 
-var cleanArgs = append(os.Args)
+var cleanArgs = append([]string(nil), os.Args...)
```
Comment (TomerHeber) [inline terratag_test.go:22]: Go staticheck was complaing about this.
The newer version is less ambiguous.

File: terratag_test.go [hunk 4]
```diff
@@ -127,17 +127,17 @@ func itShouldGenerateExpectedTerratagFiles(suiteDir string, g *GomegaWithT) {
 }
 
 func itShouldRunTerraformValidate(entryDir string, g *GomegaWithT) {
-	err := terraform(entryDir, "validate")
+	err := run_terraform(entryDir, "validate")
```
Comment (TomerHeber) [inline terratag_test.go:130]: name conflicts...
modifed terraform function to run_terraform and terratag function to run_terratag.

File: .github/workflows/release.yml (hunk 1 above)
Comment (GiliFaroEnv0) [reply .github/workflows/release.yml:50]: did you run this command localy ?

File: main.go [hunk 5]
```diff
@@ -1,35 +0,0 @@
...
-	if isMissingArg {
-		return
-	}
-	initLogFiltering(args.Verbose)
-
-	cli.Terratag(args)
-}
-
-func initLogFiltering(verbose bool) {
-	level := "INFO"
```
Comment (GiliFaroEnv0) [inline main.go:23]: why this file deleted?

**Review by GiliFaroEnv0 — COMMENTED** [review]
Looks good
left some Q

File: .github/workflows/release.yml (hunk 1 above)
Comment (TomerHeber) [reply .github/workflows/release.yml]: Good point!
I actually found a problem see:
d48e230ba7262eff734776917c4adc383a0e3b69

File: main.go (hunk 5 above)
Comment (TomerHeber) [reply main.go]: Thanks for catching this!!
it was omitted due to .gitignore.
I fixed .gitingore.
It's now under ./cmd

bf20b8175cd2b3c5522387396afd97baa67587f7

**Pull request merged**